tests/Unit/PrestaShopBundle/Security/OAuth2/OAuth2ClientTest.php

```diff
@@ -57,10 +53,20 @@ public function setUp(): void
      */
     public function testGetUser(ServerRequestInterface $request, bool $exists): void
     {
-        $user = $this->client->getUser($request);
         if ($exists) {
+            $this->leagueResourceServer->method('validateAuthenticatedRequest')->willReturnCallback(function (ServerRequestInterface $request) {
+                return $request->withAttribute('oauth_client_id', $request->getParsedBody()['client_id'] ?? null);
+            });
+            $user = $this->client->getUser($request);
+
             $this->assertTrue($user instanceof UserInterface);
         } else {
+            $this
+                ->leagueResourceServer
+                ->method('validateAuthenticatedRequest')
+                ->willThrowException(new OAuthServerException('testMessage', 0, 401))
```

Comment:
```suggestion
                ->willThrowException(new OAuthServerException('Client authentication failed', 4, 'invalid_client', 401))
```

Reply:
Copied from internal code of PHPleague so you're sure these are proper arguments